src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -1198,7 +1198,10 @@ CompactionAggregate.UnifiedAggregate constructAggregate(Controller controller, l
             int threshold = level.threshold;
             int fanout = level.fanout;
             int index = level.index;
-            int maxSSTablesToCompact = Math.max(fanout, (int) Math.min(spaceAvailable / level.avg, controller.maxSSTablesToCompact()));
+            int maxSSTablesToCompact = Math.max(fanout, (int) (spaceAvailable / level.avg));
+            if (maxSSTablesToCompact > controller.maxSSTablesToCompact()) {
```

Comment:
Keep the min and just put it on the outside?

Reply:
> Keep the min and just put it on the outside?

I have applied this suggestion and added a comment

Reply:
I don't think the fanout is the issue here.  The fanout is based on the scaling parameter `w` using this formula:
`if w < 0 then f = 2 - w`
`if w > 0 then f = 2 + w`
`if w = 0 then f = 2`
The scaling parameter is usually a pretty small number (between -36 and 36) so it's not likely the fanout will ever be large enough to cause an issue.

Additionally, when the count is less than the fanout, we disregard the `max_sstables_to_compact` even with this change: https://github.com/datastax/cassandra/blob/e141d979503873ca391a1288f54de9c68b9d5b90/src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java#L1206